SpatialConvolution.lua

```diff
@@ -361,24 +184,26 @@ local function makeContiguous(self, input, gradOutput)
 end
 
 function SpatialConvolution:updateOutput(input)
-    if not self.weightDesc then self:resetWeightDescriptors() end
     input = makeContiguous(self, input)
     self:createIODescriptors(input)
-
+    local finder = find.get()
+    local fwdAlgo = finder:forwardAlgorithm(self, { self.iDesc[0], self.input_slice, self.weightDesc[0],
+                                                    self.weight, self.convDesc[0], self.oDesc[0], self.output_slice})
+    local extraBuffer, extraBufferSize = cudnn.getSharedWorkspace()
     for g = 0, self.groups - 1 do
-        errcheck('cudnnConvolutionForward', cudnn.getHandle(),
+        errcheck(self,'cudnnConvolutionForward', cudnn.getHandle(),
```

Comment:
having two different errcheck functions with the same name is confusing, can you give it a different name?